format/FlightSql.proto

```diff
@@ -1797,6 +1797,15 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+// An optional response returned when `DoPut` is called with `CommandPreparedStatementQuery`
+message DoPutPreparedStatementResult {
+  option (experimental) = true;
+
+  // (potentially updated) opaque handle for the prepared statement on the server.
+  // All subsequent requests for this prepared statement must use this new handle, if specified
```

Comment:
I think it might help to give a hint here about what this field is used for

```suggestion
  // All subsequent requests for this prepared statement must use this new handle, if specified.
  // The updated handle allows implementing query parameters with stateless services
  // as described in https://github.com/apache/arrow/issues/37720
```

Reply:
> It would be good to copy/summarize the justification here, if we think it's worth keeping, instead of/along with referencing the ticket

I've updated the docs to provide some background on this change, as well as explicitly documenting legacy behavior. also made the updated handle explicitly `optional` to help indicate that the older stateful pattern can still be supported even in up-to-date clients and servers.